testsuite/MDAnalysisTests/coordinates/test_fhiaims.py

```diff
@@ -23,32 +23,159 @@
+
+    filename, trajectory, topology = [FHIAIMS] * 3
+    reader, writer = FHIAIMSReader, FHIAIMSWriter
+    pos_atom1 = np.asarray(
+        [6.861735,   2.103823,  37.753513], dtype=np.float32)
+    dimensions = np.asarray(
+        [18.6, 18.6,  55.8, 90., 90., 90.], dtype=np.float32)
+    n_atoms = 6
+    n_frames = 1
+    time = 0.0
+    ext = '.in'
+    prec = 6
+    container_format, changing_dimensions = [False] * 2
+
+
+class TestFHIAIMSReader(object):
+
+    prec = 6
+
+    @pytest.fixture
+    def ref(self):
+        return RefFHIAIMS()
+
+    @pytest.fixture
+    def universe(self):
+        return mda.Universe(FHIAIMS, FHIAIMS)
+
+    @pytest.fixture
+    def universe2(self):
+        return mda.Universe(FHIAIMS)
+
+ #   def setUp(self):
+ #       self.ref = RefFHIAIMS
+ #       self.prec = 6
+
+    def test_single_file(self, universe, universe2):
+        assert_almost_equal(universe.atoms.positions, universe2.atoms.positions,
+                            self.prec, "FHIAIMSReader failed to load universe from single file")
+
+    def test_uses_FHIAIMSReader(self, universe):
+        from MDAnalysis.coordinates.FHIAIMS import FHIAIMSReader
+
+        assert isinstance(universe.trajectory,
+                          FHIAIMSReader), "failed to choose FHIAIMSReader"
+
+    def test_dimensions(self, ref, universe):
+        assert_almost_equal(
+            ref.dimensions, universe.dimensions,
+            self.prec, "FHIAIMSReader failed to get unitcell dimensions")
+
+    def test_n_atoms(self, ref, universe):
+        assert_equal(
+            ref.n_atoms, universe.trajectory.n_atoms,
+            "FHIAIMSReader failed to get the right number of atoms")
+
+    def test_fhiaims_positions(self, ref, universe):
+        # first particle
+        assert_almost_equal(ref.pos_atom1,
+                            universe.atoms.positions[0],
+                            self.prec,
+                            "FHIAIMSReader failed to read coordinates properly")
+
+    def test_n_frames(self, ref, universe):
+        assert_equal(ref.n_frames, universe.trajectory.n_frames,
+                     "wrong number of frames")
+
+    def test_time(self, ref, universe):
+        assert_equal(ref.time, universe.trajectory.time,
+                     "wrong time of the frame")
+
+    def test_bad_input1(self, tmpdir):
+        fname = str(tmpdir.mkdir("FHIAIMSReader").join('bad_input1.in'))
+        with open(fname, 'w') as fd:
+            fd.write('lattice_vector 1.0 0.0 0.0\n')
+            fd.write('lattice_vector 0.0 1.0 0.0\n')
+            fd.write('atom 0.1 0.1 0.1 H')
+        try:
+            u = mda.Universe(fname)
+        except Exception as e:
+            if 'Found partial periodicity' not in str(e):
+                raise ValueError(
+                    "FHIAIMSReader did not detect a missing lattice_vector")
+
+    def test_bad_input2(self, tmpdir):
+        fname = str(tmpdir.mkdir("FHIAIMSReader").join('bad_input2.in'))
```

Comment:
Make it a StringIO fixture `bad_input_illegal_line` or something like that (words are better than numbers).

Reply:
done